test/php/gutenberg-coding-standards/Gutenberg/Sniffs/Commenting/FunctionCommentSinceTagSniff.php

```diff
@@ -48,90 +91,444 @@ public function process( File $phpcsFile, $stackPtr ) {
+			$phpcs_file->addError( $missing_since_tag_error_message, $stack_pointer, $violation_code );
 			return;
 		}
 
-		// The sniff intentionally doesn't check if the docblock has a valid open tag.
-		// Its only job is to make sure that the @since tag is present and has a valid version value.
-		$doc_block_start_token = $phpcsFile->findPrevious( Tokens::$commentTokens, ( $doc_block_end_token - 1 ), null, true, null, true );
-		if ( false === $doc_block_start_token ) {
-			$phpcsFile->addError( $missing_since_tag_error_message, $stackPtr, 'MissingSinceTag' );
+		list( $doc_block_start_token, $doc_block_end_token ) = $docblock;
+
+		$version_tags = static::parse_since_tags( $phpcs_file, $doc_block_start_token, $doc_block_end_token );
+		if ( empty( $version_tags ) ) {
+			$phpcs_file->addError( $missing_since_tag_error_message, $stack_pointer, $violation_code );
 			return;
 		}
 
-		// This is the first non-docblock token, so the next token should be used.
-		++$doc_block_start_token;
+		foreach ( $version_tags as $since_tag_token => $version_value_token ) {
+			if ( null === $version_value_token ) {
+				$phpcs_file->addError( $missing_since_tag_error_message, $since_tag_token, $violation_code );
+				continue;
+			}
+
+			$version_value = $tokens[ $version_value_token ]['content'];
+
+			if ( static::validate_version( $version_value ) ) {
+				continue;
+			}
+
+			$phpcs_file->addError(
+				'Invalid @since version value for the "%s" %s. Version value must be greater than or equal to 0.0.1.',
+				$version_value_token,
+				'Invalid' . $capitalized_token_type . 'SinceTagVersionValue',
+				array(
+					$token_name,
+					$version_value,
+				)
+			);
+		}
+	}
 
-		$since_tag_token = $phpcsFile->findNext( T_DOC_COMMENT_TAG, $doc_block_start_token, $doc_block_end_token, false, '@since', true );
-		if ( false === $since_tag_token ) {
-			$phpcsFile->addError( $missing_since_tag_error_message, $stackPtr, 'MissingSinceTag' );
+	/**
+	 * Processes a token representing an object-oriented property to check for a missing @since tag in its docblock.
+	 *
+	 * @param File $phpcs_file    The file being scanned.
+	 * @param int  $stack_pointer The position of the object-oriented property token in the stack.
+	 */
+	protected function process_property_token( File $phpcs_file, $stack_pointer ) {
+		$tokens = $phpcs_file->getTokens();
+
+		$property_name                   = $tokens[ $stack_pointer ]['content'];
+		$oo_token                        = Scopes::validDirectScope( $phpcs_file, $stack_pointer, Collections::ooPropertyScopes() );
+		$class_name                      = ObjectDeclarations::getName( $phpcs_file, $oo_token );
+		$missing_since_tag_error_message = sprintf(
+			'@since tag is missing for the "%s::%s" property.',
+			$class_name,
+			$property_name
+		);
+
+		$violation_code = 'MissingPropertySinceTag';
+
+		$scope_modifier = Variables::getMemberProperties( $phpcs_file, $stack_pointer )['scope'];
+		if ( ( $scope_modifier === 'protected'
```

Comment:
This check is very similar to a check in `process_function_token()`. Does it make sense to consolidate both in a helper method?

Reply:
Fixed in febac6fa6ea7f03eb311710add7b5566f6f04598